### PR TITLE
Add mouse utility methods

### PR DESCRIPTION
**Pull request by monkey0506**
Includes Mouse.SelectPreviousMode and Mouse.IsModeEnabled. These functions are trivial in the engine source, but less trivial for end-user devs to implement.

### CODE REVIEW

File: Engine/ac/mouse.cpp [hunk 1]
```diff
@@ -535,8 +581,10 @@ void RegisterMouseAPI()
     ccAddExternalStaticFunction("Mouse::EnableMode^1",              Sc_enable_cursor_mode);
     ccAddExternalStaticFunction("Mouse::GetModeGraphic^1",          Sc_Mouse_GetModeGraphic);
     ccAddExternalStaticFunction("Mouse::IsButtonDown^1",            Sc_IsButtonDown);
+    ccAddExternalStaticFunction("Mouse::IsModeEnabled^1",           Sc_IsModeEnabled);
     ccAddExternalStaticFunction("Mouse::SaveCursorUntilItLeaves^0", Sc_SaveCursorForLocationChange);
     ccAddExternalStaticFunction("Mouse::SelectNextMode^0",          Sc_SetNextCursor);
+    ccAddExternalStaticFunction("Mouse::SelectPreviousMode^0",      Sc_SetPreviousCursor);
```
Comment (rofl0r) [inline Engine/ac/mouse.cpp]: why is the enduser func called "SelectNextMode" and the internal func "SetNextCursor" ? these 2 terms seem to have an entirely distinct meaning.

Comment (ivan-mogilko) [inline Engine/ac/mouse.cpp]: This is a legacy issue, in AGS "cursors" and "modes" are strictly tied components of same entity (the graphical and logical ones), the type is even called "CursorMode", which can refer both to chosen interaction mode and its cursor.
The cursor has sprite attached to it, which may be swapped, but cursor is always same for the given mode, so in practice saying "next cursor" is same as "next mode".

**Comment by ivan-mogilko** [comment]
@monkey0506 , I would really like to ask you to move the duplicating code from **find_next_enabled_cursor** and **find_previous_enabled_cursor** into the separate function. I am talking about the code that tests if the mouse cursor can be selected:

<pre>
        if ((game.mcurs[testing].flags & MCF_DISABLED)==0) {
            // inventory cursor, and they have an active item
            if (testing == MODE_USE) 
            {
                if (playerchar->activeinv > 0)
                    break;
            }
            // standard cursor that's not disabled, go with it
            else if (game.mcurs[testing].flags & MCF_STANDARD)
                break;
        }
</pre>

**Comment by monkey0506** [comment]
@ivan-mogilko Done. I also slightly modified my implementation of `IsModeEnabled` to reflect that `eModeUseinv` is disabled while the player has no active inventory.

**Pull request merged**